src/components/annotations/draw.js

```diff
@@ -73,6 +73,26 @@ function drawOne(gd, index) {
     drawRaw(gd, options, index, false, xa, ya);
 }
 
+// Convert pixels to the coordinates relevant for the axis referred to. For
+// example, for paper it would convert to a value normalized by the dimension of
+// the plot.
+// axDomainRef: if true and axa defined, draws relative to axis domain,
+// otherwise draws relative to data (if axa defined) or paper (if not).
+function p2rR2p(axa, optAx, dAx, gsDim, vertical, axDomainRef) {
```

Comment:
If I'm understanding this correctly, you aren't _converting_ per se, you're shifting a position value (which can be an axis data value, an axis domain fraction, or a paper fraction) to a new position, offset by `dAx` pixels, right? Then I'd suggest renaming the function and args something like:

`function shiftPosition(ax, pos, pxShift, gsDim, vertical, axDomainRef)`

Reply:
:+1: